### PR TITLE
Fix: app randomly hangs when the upload queue is full

### PR DESCRIPTION
**Pull request by luguina**
This not a definitive fix but a proper workaround to ensure the app is not getting randomly frozen when the upload queue is full.
There is a race condition in the way the app execution snapshots are managed. We will review/improve the whole snapshots process but in the meanwhile, this should keep the client from getting frozen.

### CODE REVIEW

File: src/com/sheepit/client/Log.java [hunk 1]
```diff
@@ -55,15 +55,28 @@ public void error(String msg_) {
...
+					this.checkpoints.get(this.lastCheckPoint).add(line);
+				}
+				if (this.printStdOut == true) {
+					System.out.println(line);
+				}
 			}
 		}
+		catch (Exception e) {
+			// I don't want to concatenate
+			System.err.println("Log::append ERROR writing log information. Values:");
```
Comment (laurent-clouet) [inline src/com/sheepit/client/Log.java:73]: If we are running the oneline ui, we should not print in stdout/stderr.

Comment (luguina) [reply src/com/sheepit/client/Log.java]: We currently have 10+ printouts of the glitch from SheepIt users and in all the cases, it follows the same pattern. While we haven't discovered yet the root cause of the problem, printing out the error doesn't provide any additional value for us at this time.

So, the proposal is to remove any information and letting the thread continue until we finish the new Log class that manages the snapshots (checkpoints) in a completely different way.

**Comment by kryztoval** [comment]
I just had this happen to me on 2 of the 4 rendering instances in 2 different computers... and it won't upload the queued frames, even if I tell it to exit after current frame.

This means I just completed but coulnd't send 6 rendered frames.

I am using windows client 6.2040.0 11 hours waiting for it to upload them.

**Comment by Raimund58** [comment]
After two hours the frames will be reassigned to another machine. It is a hard reset.

**Comment by kryztoval** [comment]
Yeah, but the client will not try to upload them, nor will it render anything else, so it is basically at an impass and I need to restart it. For a program that is basically meant to be ran unsupervised, this dead lock is not so good. I think I will create an issue if I see this again.

**Comment by luguina** [comment]
I have a client with all the possible traces running 24/7 and I've discovered the problem. Some projects over the last week have rendered a frame that cannot be loaded by the Java library and the frame preview creates an exception that is not propagated/caught properly.

I don't have enough information in the logs to see if the problem is a huge rendered image size or some kind of error in the format or something else, but I have seen a way of not leaving the client frozen with the upload.

I'll fix it tonight and will push a quick fix.

**Pull request merged**